### PR TITLE
While-Assign-Compare

### PR DESCRIPTION
**Pull request by TheDan64**
TL;DR

Limonite-LLVM support for:
* While loops
* Assignments
* Add, Subtract, Multiply, Divide, Equal, Less Than, Less Than Equal, Greater Than, Greater Than Equal
    * Unsigned ints only for now. Need to work out how to determine if signed
    * Floats (u32, u64, u128). LLVM has a U and O type. Not sure yet what the difference is. Did O type.
* Basic Auto deref support. IE integers become pointers to ints and operators deference them. So `var i = 1` is really `u8 * i = 1` and (limonite) `i + j` is really (llvm) `*i + *j` and gets dereferenced to `i + j`. Definitely needs a lot more work, but provides a good start.

LLVM-lib support:
* Various builder functions such as free, xor, and, or, get_type, get_bit_size, etc
* Helper functions such as Value.is_pointer(), is_int(), is_float(), etc
* Appending basic blocks after basic blocks (surprisingly, doesn't exist in LLVM but the pieces to do it does)
* Some of these might be simplified later. Once I can distinguish signed from unsigned ints I can probably make generic methods that work differently for signed/unsigned/float inputs
* Verified some functions that I speculated should be Options due to null pointers and made them such

Unit Tests
* Three new ones: test_while_gt_decrement_u8, test_while_lt_increment_u8, test_hello_world
* Macros!! Reduced a lot of test boilerplate by adding macros to simplify AST writing and management in LLVM unit tests

### CODE REVIEW

**Comment by TheDan64** [comment]
Did [more research](https://stackoverflow.com/questions/8627331/what-does-ordered-unordered-comparison-mean) and Float O vs U is ordered vs unordered. Basically just how it handles NaN comparisons. Unordered seems to always return true when NaN is involved (Unless it's NaN vs NaN in which case the opposite seems true). I think it makes more sense to stick to ordered, then. Which is also what [Rust](https://is.gd/Lkumvf) & [Python](https://repl.it/HtTW/1) seem to do.
